### PR TITLE
[CIS-735] Add convenient ChatClient extensions to instantiate VCs

### PR DESCRIPTION
**Pull request by cardoso**
Adds extensions in StreamChatUI for ChatClient to instantiate ChatChannelListVC and ChatChannelVC more easily.

### CODE REVIEW

File: Sources/StreamChatUI/Utils/ChatClient+VCs.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        channelListVC.controller = controller
+        return channelListVC
+    }
+    
+    func channelVC(type: ChannelType, id: String) -> _ChatChannelVC<ExtraData> {
+        let channelVC = _ChatChannelVC<ExtraData>()
+        let channelId = ChannelId(type: type, id: id)
+        let controller = channelController(for: channelId)
+        channelVC.channelController = controller
+        return channelVC
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/Utils/ChatClient+VCs.swift:21]: Why not just `func channelVC(cid: ChannelId) -> _ChatChannelVC<ExtraData>` ? We use `cid` as the main chat identifier everywhere anyway.

**Review by VojtaStavik — COMMENTED** [review]
This is a good start. Can you please add documentation (it's public API) and tests (to verify the VC's have the controllers with correct parameters).

File: Sources/StreamChatUI/Utils/ChatClient+VCs.swift [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import StreamChat
+
+public extension _ChatClient {
+    func channelListVC(query: _ChannelListQuery<ExtraData.Channel>? = nil) -> _ChatChannelListVC<ExtraData> {
+        let query = query ?? .init(filter: .containMembers(userIds: [currentUserId].compactMap { $0 }))
+        let channelListVC = _ChatChannelListVC<ExtraData>()
+        let controller = channelListController(query: query)
+        channelListVC.controller = controller
+        return channelListVC
+    }
```
[the remark this answers is not in the document]
Comment (cardoso) [reply Sources/StreamChatUI/Utils/ChatClient+VCs.swift:14]: I think we should at least create a .containsCurrentUser kind of filter. What do you think?

**Comment by cardoso** [comment]
Changes made!

**Pull request closed without merging**